cluster-autoscaler/cloudprovider/gce/gce_cloud_provider.go

```diff
@@ -137,15 +138,36 @@ func (gce *GceCloudProvider) GetAvailableMachineTypes() ([]string, error) {
 
 // NewNodeGroup builds a theoretical node group based on the node definition provided. The node group is not automatically
 // created on the cloud provider side. The node group is not returned by NodeGroups() until it is created.
-func (gce *GceCloudProvider) NewNodeGroup(machineType string, labels map[string]string, extraResources map[string]resource.Quantity) (cloudprovider.NodeGroup, error) {
+func (gce *GceCloudProvider) NewNodeGroup(machineType string, labels map[string]string, systemLabels map[string]string,
+	extraResources map[string]resource.Quantity) (cloudprovider.NodeGroup, error) {
 	nodePoolName := fmt.Sprintf("%s-%s-%d", nodeAutoprovisioningPrefix, machineType, time.Now().Unix())
+	zone := gce.gceManager.getLocation()
+
+	if gpuRequest, found := extraResources[gpu.ResourceNvidiaGPU]; found {
+		gpuType, found := systemLabels[gpu.GPULabel]
+		if !found {
+			return nil, cloudprovider.ErrIllegalConfiguration
```

Comment:
Why the system label is needed?

Reply:
OH is see - you always try to pass it and set to default if not present.

Reply:
Because extraResources has an information that we want a GPU, not what GPU we want. Note that we set this label even if there was no nodeSelector on any user pods.